iocore/net/quic/QUICPacketReceiveQueue.cc

```diff
@@ -133,6 +133,15 @@ QUICPacketReceiveQueue::dequeue(QUICPacketCreationResult &result)
       this->_offset      = 0;
     }
   } else {
+    if (!this->_packet_factory.get_hs_protocol()->is_handshake_finished() && udp_packet) {
```

Comment:
I'd rather add `is_ready_to_create_protected_packet()` instead of exposing QUICHandshakeProtocol. I think `set_hs_protocol` should be removed in the future, because QUICPacketFactory only cares key availability.

Reply:
make sense, I refined the fix.